src/lib.js

```diff
@@ -3,27 +3,64 @@
 const shopUrl = require('generate-db-shop-urls')
 const { DateTime } = require('luxon')
 
-const fallbackLink = (journey, bahncard, travelClass) => {
-	const trainLegs = journey.legs.filter(l => l.mode !== 'walking')
-	const origin = trainLegs[0].origin
-	const departure = trainLegs[0].departure
-	const destination = trainLegs[trainLegs.length - 1].destination
-	const date = DateTime.fromISO(departure).setZone('Europe/Berlin').toFormat('dd.MM.yy')
-
-	const data = {
-		startSucheSofort: true,
-		startBhfName: origin.name,
-		startBhfId: '00' + origin.id,
-		zielBhfName: destination.name,
-		zielBhfId: '00' + destination.id,
-		schnelleVerbindungen: true,
-		klasse: travelClass,
-		tripType: 'single',
-		datumHin: date,
-		travellers: [{ typ: 'E', bc: bahncard }],
-	}
+const ageMap = { // gotten by hand from r (first param)
+	'A': 13,
+	'Y': 11,
+}
+
+// https://www.bahn.de/web/api/angebote/stammdaten/ermaessigungsarten
+const discounts = require('./discounts.json')
+
+
+const bcMap = {
+	'0': 'k.E.',
+	'2': 'BC25',
+	'4': 'BC50',
+}
+
+const dcMap = {}
+
+for (const discount of discounts) {
+	dcMap[discount.abkuerzung] = discount
+}
+
+const fallbackLink = async (journey, bahncard, travelClass, age = 'A') => {
+	if (!['0', '1', '2', '3', '4'].includes(bahncard)) throw new Error('invalid bahncard')
+	if (!['1', '2'].includes(travelClass)) throw new Error('invalid class')
+	if (journey.type !== 'journey' || !journey.legs || journey.legs.length === 0) throw new Error('invalid journey')
+	if (!['A', 'Y'].includes(age)) throw new Error('invalid age')
+
+	const params = new URLSearchParams()
+	params.set('sts', 'true')
+
+	const firstLeg = journey.legs[0]
+	params.set('so', firstLeg.origin.name) // display
+	params.set('soid', `A=1@L=${firstLeg.origin.id}@`) // hafas id
+
+	const lastLeg = journey.legs[journey.legs.length - 1]
+	params.set('zo', lastLeg.destination.name) // display
+	params.set('zoid', `A=1@L=${lastLeg.destination.id}@`) // hafas id
+
+	params.set('kl', travelClass)
+
+	params.set('hd', firstLeg.plannedDeparture) // from when
+	params.set('hza', 'D') // D = departure, A = arrival
+
+	const discountAbkuerzung = bcMap[bahncard]
+	const discount = dcMap[discountAbkuerzung]
+	const klasse = discount.klassen.includes('KLASSENLOS') ? 'KLASSENLOS' : 'KLASSE_' + travelClass
+
+	params.set('r', `${ageMap[age]}:${discount.id}:${klasse}:1`)
+
+	// no idea what those mean
```

Comment:
- ar = seat reservation only
- s = prefer fast connections
- d = direct connections only
- hz = intermediate stops (up to two entries, formatted as: `["<hafas id>","<display name>",<minutes overlay>]`)
- fm = taking bike on train
- bp = best price search

I researched this by using `https://www.bahn.de/buchung/start?` and adding the appropriate parameters, then comparing the toggles and options shown.